### PR TITLE
Adds picture frames

### PR DESCRIPTION
**Pull request by Tastyfish**
This PR adds glass and wooden picture frames for papers, photos, posters, and canvases.
* Based on [this thread](http://nanotrasen.se/phpbb3/viewtopic.php?f=12&t=7143).
* Supports both landscape and portrait posters, as well as all 4 canvas sizes.
* Clicking the frame allows you to view photos and papers, as normal.
* Glass frames are made in the autolathe.
* Wood frames are made with a plank of wood.
* Rumors say that the design of the hanger allows certain devices to be attached to the back...
 * (When the frame is tilted, you can attach grenades or C4. It takes 15 seconds and people can see you doing it.).
* To use it...
 1. Click the frame with a paper, photo, or poster.
 2. Click the filled frame on a wall.
* To take it apart...
 1. Screwdriver it to get it off the wall. This takes 10 seconds.
 2. Screwdriver the loose frame to get the thing out. (You can skip this step and immediately crowbar at this point.)
 3. Crowbar it to dismantle. 

Frames
Frames 2

:cl:
rscadd: There are now picture frames that can be made from the autolathe or wood planks for papers, photos, posters, and canvases!
/:cl:

### CODE REVIEW

**Comment by Regen1** [comment]
So I can frame my ass now?

:+1:  :100:

**Comment by Tastyfish** [comment]
Yes

**Comment by NTSAM** [comment]
:ok_hand:

**Comment by Tastyfish** [comment]
Dangit now I actually want to do that silly assembly thing.

**Comment by Dumbdumn5** [comment]
:+1:

**Comment by KasparoVy** [comment]
Make it work with grenades, the little pop-can IEDs, and C4! Now wouldn't that be interesting.

**Comment by Tastyfish** [comment]
Added the attaching nades/c4 thing.

**Comment by Regen1** [comment]
Ahahah i love it

**Comment by fludd12** [comment]
Does this / is this planned to support the custom canvasses?

**Comment by Gummin** [comment]
Awesome.

**Comment by FlattestGuitar** [comment]
:+1: x :100: 
Finally I can hang someone's ass on a wall in a _classy_ way.

**Comment by NTSAM** [comment]
Yay, now I can booby trap my worst enemy in the cruelest possible way.

**Comment by Tastyfish** [comment]
Oh right, canvases haha, I'll add that.

**Comment by sasanek12** [comment]
:+1:
No more tearing off my Communist Propaganda

**Comment by GentlemanlyHeadcrab** [comment]
What about standing pictures that I can sit at a desk and pick up dramatically?

**Comment by fludd12** [comment]
[Leaves poster in frame: "Remember, you're here forever"]

**Comment by TullyBurnalot** [comment]
:+1:

**Comment by Tastyfish** [comment]
You can use them with canvases now; screenshot added. I also fixed a couple weird things with the canvas code. (Although. the bug where you have to mess with it to make it render for everyone else is probably still there -- it's nearly impossible to test. However, putting it in a frame should make it render either way.)

**Comment by FalseIncarnate** [comment]
I spy dickbutt and someone taking my idea of using the vacant office as an art gallery in the same preview.

:+1:

**Comment by CrAzYPiLoT-SS13** [comment]
Yeah, I would love to keep a standing picture of wet skr.... my family on
my desk.

**Comment by Tastyfish** [comment]
You can technically place them on tables as items; just imagine they're standing within the logic of ss13's visuals. :wink:

File: code/modules/paperwork/frames.dm [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+	if(D)
+		insert(D)
+	update_icon()
+
+/obj/item/weapon/picture_frame/Destroy()
+	if(displayed)
+		displayed = null
+		for(var/A in contents)
+			qdel(A)
+	..()
```
Comment (Fox-McCloud) [inline code/modules/paperwork/frames.dm]: return ..()

File: code/modules/paperwork/frames.dm [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+		transform = tilt_transform
+		verbs |= /obj/structure/sign/picture_frame/proc/untilt
+	else
+		verbs |= /obj/structure/sign/picture_frame/proc/tilt
+
+/obj/structure/sign/picture_frame/Destroy()
+	if(frame)
+		qdel(frame)
+		frame = null
+	..()
```
Comment (Fox-McCloud) [inline code/modules/paperwork/frames.dm]: `return ..()`

File: code/modules/paperwork/frames.dm (hunk 1 above)
Comment (Tastyfish) [inline code/modules/paperwork/frames.dm]: Whoops, good catch.

**Pull request merged**